resource/upgrade_identity.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package resource
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+)
+
+// Request information for the provider logic to update a resource identity
+// from a prior resource identity version to the current identity version.
+type UpgradeResourceIdentityRequest struct {
+	// TypeName is the type of resource that Terraform needs to upgrade the
+	// identity state for.
+	TypeName string
+
+	// Version is the version of the identity state the resource currently has.
+	Version int64
+
+	// RawIdentity is the identity state as Terraform sees it right now in JSON. See the
+	// documentation for `RawIdentity` for information on how to work with the
+	// data it contains.
```

Comment:
Might want to mention that this is meant for providers wanting to skip redeclaring old schemas and that it's considered advanced, see: https://github.com/hashicorp/terraform-plugin-framework/blob/17f1faf8787c6987ade7b6d58e297f256473ab43/resource/upgrade_state.go#L17-L26

Also this comment mentions looking at other documentation, but I'm not sure what that meant, perhaps left over from elsewhere?